fclient/federationclient.go

```diff
@@ -191,8 +203,8 @@ func makeVersionQueryString(roomVersions []gomatrixserverlib.RoomVersion) string
 // See https://matrix.org/docs/spec/server_server/unstable.html#joining-rooms
 func (ac *federationClient) MakeJoin(
 	ctx context.Context, origin, s spec.ServerName, roomID, userID string,
-	roomVersions []gomatrixserverlib.RoomVersion,
 ) (res RespMakeJoin, err error) {
+	roomVersions := roomVersionsToList(gomatrixserverlib.StableRoomVersions())
```

Comment:
By using stable room versions we make it impossible to make_join with unstable room versions e.g MSCs, and there seems to be no way to say "no really, please do".